src/Reflex/Dom/Builder/Static.hs

```diff
@@ -0,0 +1,168 @@
+  (result, a') <- runStateT a []
+  return (result, mconcat $ reverse a')
+
+instance MonadReflexCreateTrigger t m => MonadReflexCreateTrigger t (StaticDomBuilderT t m) where
+  {-# INLINABLE newEventWithTrigger #-}
+  newEventWithTrigger = lift . newEventWithTrigger
+  {-# INLINABLE newFanEventWithTrigger #-}
+  newFanEventWithTrigger f = lift $ newFanEventWithTrigger f
+
+instance PerformEvent t m => PerformEvent t (StaticDomBuilderT t m) where
+  type Performable (StaticDomBuilderT t m) = Performable m
+  {-# INLINABLE performEvent_ #-}
+  performEvent_ e = lift $ performEvent_ e
+  {-# INLINABLE performEvent #-}
+  performEvent e = lift $ performEvent e
+
+instance MonadSample t m => MonadSample t (StaticDomBuilderT t m) where
+  {-# INLINABLE sample #-}
+  sample = lift . sample
+
+instance MonadHold t m => MonadHold t (StaticDomBuilderT t m) where
+  {-# INLINABLE hold #-}
+  hold v0 v' = lift $ hold v0 v'
+  {-# INLINABLE holdDyn #-}
+  holdDyn v0 v' = lift $ holdDyn v0 v'
+  {-# INLINABLE holdIncremental #-}
+  holdIncremental v0 v' = lift $ holdIncremental v0 v'
+
+instance (Reflex t, Monad m, MonadHold t (StateT [Behavior t ByteString] m)) => Deletable t (StaticDomBuilderT t m) where
+  {-# INLINABLE deletable #-}
+  deletable delete (StaticDomBuilderT a) = StaticDomBuilderT $ do
+    (result, a') <- lift $ runStateT a []
+    let html = mconcat $ reverse a'
+    b <- hold html (mempty <$ delete)
+    modify (join b:)
+    return result
+
+instance (Monad m, Ref m ~ Ref IO, Reflex t) => TriggerEvent t (StaticDomBuilderT t m) where
+  {-# INLINABLE newTriggerEvent #-}
+  newTriggerEvent = return (never, const $ return ())
+  {-# INLINABLE newTriggerEventWithOnComplete #-}
+  newTriggerEventWithOnComplete = return (never, \_ _ -> return ())
+  {-# INLINABLE newEventWithLazyTriggerWithOnComplete #-}
+  newEventWithLazyTriggerWithOnComplete _ = return never
+
+type SupportsStaticDomBuilder t m = (Reflex t, MonadIO m, MonadHold t m, MonadFix m, PerformEvent t m, Performable m ~ m, MonadReflexCreateTrigger t m, Deletable t m, MonadRef m, Ref m ~ Ref IO)
+
+data StaticDomSpace
+instance DomSpace StaticDomSpace where
+    type RawElement StaticDomSpace = ByteString
+
+instance SupportsStaticDomBuilder t m => DomBuilder t (StaticDomBuilderT t m) where
+  type DomBuilderSpace (StaticDomBuilderT t m) = StaticDomSpace
+  {-# INLINABLE textNode #-}
+  textNode (TextNodeConfig contents) = StaticDomBuilderT $ do
+    modify . (:) $ constant $ BL.toStrict $ toLazyByteString $ fromHtmlEscapedText contents
+    return TextNode
+
+  {-# INLINABLE element #-}
+  element elementTag cfg child = do
+    let toAttr (_mns,k) v = let key = encodeUtf8 k; val = BL.toStrict $ toLazyByteString $ fromHtmlEscapedText v in [key <> "=\"" <> val  <> "\""]
+    let attrs0 = Map.foldMapWithKey toAttr $ cfg ^. initialAttributes
+    let attrs1 = if null attrs0 then "" else " " <> B8.intercalate " " attrs0
+
+    let tag' = encodeUtf8 elementTag
```

Comment:
OK, cool; here's the citation: https://www.w3.org/TR/html-markup/syntax.html#syntax-elements

> Tag names are used within element start tags and end tags to give the element’s name. HTML elements all have names that only use characters in the range 0–9, a–z, and A–Z.